src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -106,17 +154,30 @@ void activity_classifier::train(gl_sframe data, std::string target_column_name,
 {
   // Begin printing progress.
   // TODO: Make progress printing optional.
-  // TODO: Report accuracy.
-  // TODO: Support validation set.
-  training_table_printer_.reset(new table_printer(
-      { {"Iteration", 12}, {"Train Loss", 12}, {"Elapsed Time", 12} }));
 
-  // Instantiate the training dependencies: data iterator, compute context,
-  // backend NN model.
+  if (variant_is<gl_sframe>(validation_data) ){
+
+    training_table_printer_.reset(new table_printer(
+    { {"Iteration", 12}, {"Train Accuracy", 12}, {"Train Loss", 12}, {"Validation Accuracy", 12}, {"Validation Loss", 12}, {"Elapsed Time", 12} }));
+
+  }
+  // else if (variant_is<flex_string>(validation_data) ){
```

Comment:
Let's remove commented out code.

Reply:
Yeah I am still working on this PR. I had uploaded it so that Nick can look at it because I was getting NAN values for accuracy and loss. I was not resetting the validation data iterator. Although that is fixed now but this code needs more work. 

The reason for keeping this code was that I want to eventually accept "auto" string as well. Then elaborate on splitting the dataset itself.

I will add you for request as soon as I am done.

Reply:
Sorry @shreyajain17 - I didn't realize it was WIP. I'll make sure to check for that in the future.